ports/esp8266/network_wlan.c

```diff
@@ -170,6 +170,38 @@ static mp_obj_t esp_disconnect(mp_obj_t self_in) {
 }
 static MP_DEFINE_CONST_FUN_OBJ_1(esp_disconnect_obj, esp_disconnect);
 
+static mp_obj_t build_stations_tuple(void) {
+    mp_uint_t count = wifi_softap_get_station_num();
+    if (count == 0) {
+        return mp_const_empty_tuple;
+    }
+    mp_obj_tuple_t *stations = NULL;
+    struct station_info *info = wifi_softap_get_station_info();
+    if (info == NULL) {
```

Comment:
Can you omit this check, and then the loop below just won't execute any iterations.  Then it'll return an empty list.  I think that's fine?

Reply:
I remember I looked at the disassembly for `wifi_softap_free_station_info` and I recall that function not making any checks before deallocating memory that was meant to be allocated in `wifi_softap_get_station_info`.  

If I understood the disassembled code correctly then the NULL check must still be there, as if `wifi_softap_get_station_info` failed to allocate memory then who knows what happens later.  I'm not 100% sure on this, though, as I don't know Xtensa assembler as much as RISC-V so I might have missed something there.

Reply:
OK.  My main point here is that instead of raising an exception if `info` is `NULL`, just return an empty list.  That will lead to smaller code.

Reply:
Well, my point was that `wifi_softap_get_station_info` returning NULL even though `wifi_softap_get_station_num` returned a non-zero value is an exceptional condition and should be treated as such.

Anyway, I've modified the code as you suggested, for the sake of a smaller footprint :)